lib/upipe/uprobe_stdio.c

```diff
@@ -171,6 +179,18 @@ static int uprobe_stdio_throw(struct uprobe *uprobe, struct upipe *upipe,
 
     flockfile(s);
     while (msg != NULL && *msg != '\0') {
+        if (uprobe_stdio->time_format) {
+            time_t t = time(NULL);
+            struct tm tm;
+            struct tm *tm_p = localtime_r(&t, &tm);
+            char str[TIME_FORMAT_MAX_SIZE];
+            const char *sfx = " ";
+            if (!strftime(str, sizeof (str), uprobe_stdio->time_format, tm_p)) {
+                str[0] = '\0';
+                sfx = "";
+            }
```

Comment:
This code should be hoisted outside the loop, because all lines of a multi-line log should have the same timestamp.